packages/profile-details-nodejs/src/plugin.ts

```diff
@@ -0,0 +1,232 @@
+/* eslint-disable indent */ // buggy linting for the originalImplementation conditional signInUpPOST
+import { SuperTokensPlugin } from "supertokens-node/types";
+import { SessionContainerInterface } from "supertokens-node/recipe/session/types";
+import {
+  PLUGIN_ID,
+  PLUGIN_SDK_VERSION,
+  HANDLE_BASE_PATH,
+  SUPERTOKENS_PLUGIN_PROGRESSIVE_PROFILING_ID,
+  DEFAULT_REGISTER_SECTIONS_FOR_PROGRESSIVE_PROFILING,
+} from "./constants";
+import { getUser } from "supertokens-node";
+import { withRequestHandler } from "@shared/nodejs";
+import { createPluginInitFunction } from "@shared/js";
+import { SuperTokensPluginProfileDetailsConfig, SuperTokensPluginProfileDetailsNormalisedConfig } from "./types";
+import type { BaseFormFieldPayload } from "@supertokens-plugins/profile-details-shared";
+import { BASE_FORM_SECTIONS } from "@supertokens-plugins/profile-details-shared";
+import { enableDebugLogs, logDebugMessage } from "./logger";
+import { Implementation } from "./implementation";
+
+export const init = createPluginInitFunction<
+  SuperTokensPlugin,
+  SuperTokensPluginProfileDetailsConfig,
+  Implementation,
+  SuperTokensPluginProfileDetailsNormalisedConfig
+>(
+  (pluginConfig, implementation) => {
+    return {
+      id: PLUGIN_ID,
+      compatibleSDKVersions: PLUGIN_SDK_VERSION,
+      overrideMap: {
```

Comment:
Please check if we can make use of the `dependencies` feature of `createPluginInitFunction` to automatically add the base plugin (which we can set as a peer dep)